packages/cloudflare/src/cli/build/patches/plugins/fetch-cache-wait-until.ts

```diff
@@ -0,0 +1,44 @@
+import { getCrossPlatformPathRegex } from "@opennextjs/aws/utils/regex.js";
+
+import { patchCode } from "../ast/util.js";
+import type { ContentUpdater } from "./content-updater.js";
+
+/**
+ * The following Next.js code sets values in the incremental cache for fetch calls:
+ *  https://github.com/vercel/next.js/blob/e5fc495e3d4/packages/next/src/server/lib/patch-fetch.ts#L690-L728
+ *
+ * The issue here is that this promise is never awaited in the Next.js code (since in a standard node.js server
+ * the promise will eventually simply just run) but we do need to run it inside `waitUntil` (so that the worker
+ * is not killed before the promise is fully executed), without that this promise gets discarded and values
+ * don't get saved in the incremental cache.
+ *
+ * This function wraps the promise in a `waitUntil` call (retrieved from `globalThis.__openNextAls.getStore()`).
+ */
+export function patchFetchCacheSetMissingWaitUntil(updater: ContentUpdater) {
+  return updater.updateContent(
+    "patch-fetch-cache-set-missing-wait-until",
+    {
+      filter: getCrossPlatformPathRegex(
+        String.raw`(server/chunks/.*\.js|.*\.runtime\..*\.js|patch-fetch\.js)$`,
+        { escape: false }
+      ),
+      contentFilter: /arrayBuffer\(\)\s*\.then/,
+    },
+    ({ contents }) => patchCode(contents, rule)
+  );
+}
+
+export const rule = `
+rule:
+  kind: call_expression
+  pattern: $PROMISE
+  all:
+    - has: { pattern: "$$$_.then($$$_)", stopBy: end }
+    - has: { pattern: "Buffer.from", stopBy: end }
+    - has: { regex: "CachedRouteKind.FETCH" }
```

Comment:
Could you switch back to patterns with `any` of `CachedRouteKind.FETCH` or `$_.CachedRouteKind.FETCH`

Reply:
No, replacing `has: { regex: "CachedRouteKind.FETCH" }` with `has: { pattern: "CachedRouteKind.FETCH" }` breaks the matching 😕

I don't really understand why... 😕

Reply:
> any of CachedRouteKind.FETCH or $_.CachedRouteKind.FETCH

Reply:
> Could you switch back to patterns with `any` of `CachedRouteKind.FETCH` or `$_.CachedRouteKind.FETCH`

PS: also `$_.CachedRouteKind.FETCH` doesn't match non-minified code in which `CachedRouterKind` is accessed directly (as you can see in the [test file](https://github.com/opennextjs/opennextjs-cloudflare/pull/364/files#diff-0a93f0b32ac4487765b59be2d89d4b83e920658c7f968e7466721c394db940a6))

Reply:
That's what I said :)

Reply:
ahhhh, sorry I am not sure what I read there 😓